utils/features/features.go

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package features
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/blang/semver/v4"
+	"github.com/cilium/cilium/pkg/versioncheck"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	FeatureCNIChaining        Feature = "cni-chaining"
+	FeatureMonitorAggregation Feature = "monitor-aggregation"
+	FeatureL7Proxy            Feature = "l7-proxy"
+	FeatureHostFirewall       Feature = "host-firewall"
+	FeatureICMPPolicy         Feature = "icmp-policy"
+	FeatureTunnel             Feature = "tunnel"
+	FeatureEndpointRoutes     Feature = "endpoint-routes"
+
+	FeatureKPRMode                Feature = "kpr-mode"
+	FeatureKPRExternalIPs         Feature = "kpr-external-ips"
+	FeatureKPRGracefulTermination Feature = "kpr-graceful-termination"
+	FeatureKPRHostPort            Feature = "kpr-hostport"
+	FeatureKPRSocketLB            Feature = "kpr-socket-lb"
+	FeatureKPRNodePort            Feature = "kpr-nodeport"
+	FeatureKPRSessionAffinity     Feature = "kpr-session-affinity"
+
+	FeatureHostPort Feature = "host-port"
+
+	FeatureNodeWithoutCilium Feature = "node-without-cilium"
+
+	FeatureHealthChecking Feature = "health-checking"
+
+	FeatureEncryptionPod  Feature = "encryption-pod"
+	FeatureEncryptionNode Feature = "encryption-node"
+
+	FeatureIPv4 Feature = "ipv4"
+	FeatureIPv6 Feature = "ipv6"
+
+	FeatureFlavor Feature = "flavor"
+
+	FeatureSecretBackendK8s Feature = "secret-backend-k8s"
+
+	FeatureCNP Feature = "cilium-network-policy"
+	FeatureKNP Feature = "k8s-network-policy"
+
+	// Whether or not CIDR selectors can match node IPs
+	FeatureCIDRMatchNodes Feature = "cidr-match-nodes"
+
+	FeatureAuthSpiffe Feature = "mutual-auth-spiffe"
+
+	FeatureIngressController Feature = "ingress-controller"
+
+	FeatureEgressGateway Feature = "enable-ipv4-egress-gateway"
+)
+
+// Feature is the name of a Cilium Feature (e.g. l7-proxy, cni chaining mode etc)
+type Feature string
+
+// FeatureStatus describes the status of a Feature. Some features are either
+// turned on or off (c.f. Enabled), while others additionally might include a
+// Mode string which provides more information about in what mode a
+// particular Feature is running ((e.g. when running with CNI chaining,
+// Enabled will be true, and the Mode string will additionally contain the name
+// of the chained CNI).
+type FeatureStatus struct {
```

Comment:
Same remark about name stuttering here:

```suggestion
type Status struct {
```